Extensions/ParticleSystem/ExtensionSubDeclaration3.cpp

```diff
@@ -135,14 +135,25 @@ void ExtensionSubDeclaration3(gd::ObjectMetadata& obj) {
       .UseStandardRelationalOperatorParameters("number");
 
   obj.AddAction("Texture",
-                _("Image"),
-                _("Change the image of particles ( if displayed )."),
+                _("Change image (using an expression)"),
+                _("Change the image of particles (if displayed)."),
+                _("Change the image of particles of _PARAM0_ to _PARAM1_"),
+                _("Advanced"),
+                "CppPlatform/Extensions/particleSystemicon24.png",
+                "CppPlatform/Extensions/particleSystemicon16.png")
+      .AddParameter("object", _("Object"), "ParticleEmitter")
+      .AddParameter("string", _("Image to use"))
+      .SetParameterLongDescription("Indicate the name of the resource");
+
+  obj.AddAction("TextureAmongResources",
```

Comment:
SetTextureFromResource

(I sadly in the past not used verb at the beginning of actions. But you should have a verb now for actions, Set being the default one)